File: codewars/Sum of Digits - Digital Root/Sum of Digits - Digital Root.js
function sumDigits(num) {
    return num.toString().split('').reduce((sum, digit) => sum + Number(digit), 0);
}

function digitalRoot(num) {
    while (num >= 10) {
        num = sumDigits(num);
    }
    return num;
}

let result = digitalRoot(132189);
console.log(result);

File: codewars/Sum of Digits - Digital Root/Sum_of_Digits_Digital_Root.cpp
#include <bits/stdc++.h>

using namespace std;

int digital_root(int n) {
    int sum = 0;
    while (n > 0) {
        sum += n % 10;
        n /= 10;
    }
    if (sum >= 10) sum = digital_root(sum);
    return sum;
}